packages/cli/internal/pkg/aws/cdk/execute_cdk.go

```diff
@@ -74,11 +74,13 @@ func deleteCDKOutputDir(cdkOutputDir string) {
 	}
 }
 
-func processOutputs(stdout *bufio.Scanner, stderr *bufio.Scanner) (chan ProgressEvent, *sync.WaitGroup) {
+func processOutputs(stdout *bufio.Scanner, stderr *bufio.Scanner, uniqueKey string) (chan ProgressEvent, *sync.WaitGroup) {
 	var wait sync.WaitGroup
 	wait.Add(2)
 	progressChan := make(chan ProgressEvent)
-	currentEvent := &ProgressEvent{}
+	currentEvent := &ProgressEvent{
+		UniqueKey: uniqueKey,
```

Comment:
Rather than piping the unique key down to here, could we generate it here or just one level above? We're basically trying to tie a single CDK execution to a specific event channel so that we can differentiate between each while displaying the progress bar. Unless we need to specifically look up the channel by its key later a simple `rand.Int()` should suffice

Reply:
We don't just use it for that value, we also use it to tie the outputs together (since they could be any one of them). If you look at the function "combineProgressEvents", we create an empty object called "progressResult" which we add to a results list. This results value is then updated by the ProgressEvent stream which requires the uniqueKey value.